src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -526,6 +527,7 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 					this.shellLaunchConfig.color = defaultProfile.color;
 					this.shellLaunchConfig.env = defaultProfile.env;
 				}
+				this.shellLaunchConfig.shellIntegrationEnvironmentReporting = this._configurationService.getValue(TerminalSettingId.ShellIntegrationEnvironmentReporting);
```

Comment:
I don't think you need this since it's handled in the constructor?

Reply:
Yeap, also handled in ---> https://github.com/microsoft/vscode/pull/240424/commits/c017f3cbd2f155453c9f29f0e9ebbcb03cecae73